src/gtest/test_cumulativehash.cpp

```diff
@@ -0,0 +1,59 @@
+#include <gtest/gtest.h>
+
+#include "main.h"
+
+extern CBlockIndex* AddToBlockIndex(const CBlockHeader& block);
+extern void CleanUpAll();
+
+class ScTxCumulativeHashTestSuite: public ::testing::Test {
+
+public:
+    ScTxCumulativeHashTestSuite(){}
+
+    ~ScTxCumulativeHashTestSuite(){}
+};
+
+TEST_F(ScTxCumulativeHashTestSuite, CBlockIndexSerialization)
+{
+    CBlockIndex originalpindex;
+    originalpindex.scCumTreeHash = CPoseidonHash(libzendoomc::ScFieldElement({std::vector<unsigned char>(size_t(SC_FIELD_SIZE), 'a')}));
+
+    CDataStream ssValue(SER_DISK, PROTOCOL_VERSION);
+    ssValue << CDiskBlockIndex(&originalpindex);
+    CDiskBlockIndex diskpindex;
+    ssValue >> diskpindex;
+
+    EXPECT_TRUE(originalpindex.scCumTreeHash == diskpindex.scCumTreeHash);
```

Comment:
I really like the structure of this test. It makes it precise since you compare the whole original/retrieved objects with == but it's also robust since you do not depend on internal structure of serialization!

Reply:
Thank you!